snuba/datasets/plans/query_plan.py

```diff
@@ -50,24 +45,25 @@ class QueryPlanExecutionStrategy(ABC):
     """
 
     @abstractmethod
-    def execute(self, request: Request, runner: QueryRunner) -> QueryResult:
+    def execute(
+        self, query: Query, request_settings: RequestSettings, runner: QueryRunner,
+    ) -> QueryResult:
         """
-        Executes the query plan. The request parameter provides query and query settings.
-        The runner parameters is a function to actually run one individual query on the
+        Executes the query plan.
+        The runner parameter is a function that actually run one individual query on the
         database.
         """
         raise NotImplementedError
 
 
-class StorageQueryPlanBuilder(ABC):
+class ClickhouseQueryPlanBuilder(ABC):
```

Comment:
I think there is still a way we should pursue to preserve Clickhouse independence in some of these interfaces, but it requires to explicitly call out dataset and storages as clickhouse datasets and clickhouse storages:
- clikchouse query and clickhouse query plan stay as they are in this PR
- clickhouse query processor will have a Generic interface on top with TQueryType as parameter
- query translator will have a Generic interface on top with TQueryType
- clickhosue query plan builder will have a Generic interface on top (same parameter)
- storage will have a generic interface on top with the same parameter and all our storages will be clickhouse specific and extend from this
- dataset will be have a generic interface on top and all our dataset will be clickhouse specific.

This way we are adding some clickhosue agnostic interfaces to several components but, since we do not have a mediator component between clickhouse agnostic storage level and clickhosue specific storage level, everything from dataset down to storages have to call out clickhosue explicitly.

Reply:
Doing this here: https://github.com/getsentry/snuba/pull/904
The commit touches a lot of files which would make this PR even larger.